lib/Sema/TypeCheckDeclOverride.cpp

```diff
@@ -945,6 +1040,30 @@ bool OverrideMatcher::checkOverride(ValueDecl *baseDecl,
     emittedMatchError = true;
   }
 
+  auto baseGenericCtx = baseDecl->getAsGenericContext();
+  auto derivedGenericCtx = decl->getAsGenericContext();
+
+  // If the generic signatures are different, then complain.
+  auto newSig = getNewGenericSignature(baseDecl, decl);
+
+  if (newSig && derivedGenericCtx) {
+    auto derivedSig = derivedGenericCtx->getGenericSignature();
+
+    if (derivedSig) {
+      auto satOne = derivedSig->requirementsNotSatisfiedBy(newSig).empty();
+      auto satTwo = newSig->requirementsNotSatisfiedBy(derivedSig).empty();
```

Comment:
it occurred to me that it might be better to do `newSig->getCanonicalSignature() == derivedSig->getCanonicalSignature()`

Reply:
Ignore this part, it only makes sense if we're doing an exact comparison here, but we really just want satOne. If satTwo is false, the code is still valid but you need to thunk.